### PR TITLE
Make migration thread count counfigurable

### PR DESCRIPTION
**Pull request by gally47**
## Motivation and Context

fixes https://github.com/Graylog2/graylog2-server/issues/19792

## Screenshots (if appropriate):
Screenshot 2024-07-01 at 19 04 09

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/MigrateExistingData.tsx [hunk 1]
```diff
@@ -165,6 +176,17 @@ const MigrateExistingData = ({ currentStep, onTriggerStep }: MigrationStepCompon
                  value={values.allowlist}
                  onChange={(e) => handleChange(e, setFieldValue)}
                  required />
+          <Input id="threads"
+                 name="threads"
+                 label="Threads count"
+                 help="Threads count defines how many indices will be migrated parallelly (minimum 1, default 4)"
```
Comment (moesterheld) [inline graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/MigrateExistingData.tsx:182]:
```suggestion
                 help="Threads count defines how many indices will be migrated in parallel (minimum 1, default 4)"
```

**Review by moesterheld — CHANGES_REQUESTED** [review]
Works in general. Thank you for adding.
However, I cannot remove the value in the input field using the keyboard/mouse, I can only add to it.
Also, one small textual change.

**Comment by gally47** [comment]
> Works in general. Thank you for adding. However, I cannot remove the value in the input field using the keyboard/mouse, I can only add to it. Also, one small textual change.

@moesterheld the value is not supposed to be removed you can only replace the default value (4) with any positive integer.

**Comment by moesterheld** [comment]
> > Works in general. Thank you for adding. However, I cannot remove the value in the input field using the keyboard/mouse, I can only add to it. Also, one small textual change.
> 
> @moesterheld the value is not supposed to be removed you can only replace the default value (4) with any positive integer.

yep. never mind, I retried it and it behaves consistently with all other numeric input fields in Graylog. 
I wanted to double click the input and write another value into it, however it didn't select the value so I could directly replace it. Also, backspace won't work, which feels a little weird. I think the behavior of the numeric input could be improved in general, but it's ok for now.

**Review by moesterheld — APPROVED** [review]

File: graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/MigrateExistingData.tsx [hunk 2]
```diff
@@ -91,7 +93,15 @@ const MigrateExistingData = ({ currentStep, onTriggerStep }: MigrationStepCompon
   };
 
   const handleChange = async (e: React.ChangeEvent<any>, callback: (field: string, value: any, shouldValidate?: boolean) => Promise<void | FormikErrors<RemoteReindexRequest>>) => {
-    await callback(e.target.name, e.target.value);
+    let value;
+    value = e.target.value;
+
+    if (e.target.name === 'threads') {
+      value = Number(e.target.value) < 1 ? DEFAULT_THREADS_COUNT : Number(e.target.value);
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/MigrateExistingData.tsx:100]: We could use the `getValueFromInput` from the `FormUtils`.

**Review by ousmaneo — APPROVED** [review]
It works as expected. I have a small comment.

**Review by todvora — APPROVED** [review]
LGTM!

**Pull request merged**